src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -90,8 +92,16 @@ private void GetWorkingSetLimits(out IntPtr minWorkingSet, out IntPtr maxWorking
             minWorkingSet = IntPtr.Zero;
 
             // Get the max working set size
-            Interop.libc.rlimit info = Interop.libc.getrlimit(Interop.libc.RLIMIT_Resources.RLIMIT_RSS);
-            maxWorkingSet = new IntPtr(Convert.ToInt64(info.rlim_cur));
+            Interop.Sys.RLimit limit = new Interop.Sys.RLimit();
+            if (Interop.Sys.GetRLimit(Interop.Sys.RlimitResources.RLIMIT_RSS, ref limit) == 0)
```

Comment:
Per previous comment, I think ideally the signature here would use ```out```, and limit wouldn't need to be initialized.